UnityProject/Assets/Scripts/Player/LayDown.cs

```diff
@@ -42,9 +48,13 @@ public void ServerEnsureCorrectState()
 
 		private void CorrectState()
 		{
-			gameObject.transform.rotation = Quaternion.Euler(0, 0, 0);
-			if (disabled || health == null) return;
-			if (IsLayingDown || health.IsCrit || health.IsDead)
+			if (disabled) return;
+			var state = IsLayingDown;
```

Comment:
why create a new variable for this?

Reply:
to check for health, if health is different; affect the variable

Reply:
am I missing something? If you just used ``IsLayingDown`` seems to be exactly the same.

Reply:
can you read the lines below that

Reply:
```cs
			if (health != null)
			{
				if (health.IsDead || health.IsCrit) state = true;
			}
```

Reply:
can you not set ``IsLayingDown`` there instead of ``state``? Is it a readonly property?

Reply:
also give the if a body. Don't do things in the same line as the condition, it is confusing.

Reply:
> can you not set `IsLayingDown` there instead of `state`? Is it a readonly property?

I can, but you shouldn't change IsLayingDown because you should only use it if you're laying down. Also you'd probably crash the game by calling the server hook for the syncvar over and over again because you keep changing the variable inside the hook.

Reply:
> also give the if a body. Don't do things in the same line as the condition, it is confusing.

no, bad nesting.

Reply:
no, what you are doing is bad and is against our codeguilines.

Bad nesting is when you have multiple levels of nesting, making it confusing to grasp. This isn't bad nesting, this is just what it should be.

Reply:
It looks ugly imo, but w/e